Error building rdkafka

rdkafka fails to build due to a mismatched types error. The twitter rpc-perf repository builds successfully, but this fork does not.

### Expected behavior

Successful build. 

### Actual behavior

error[E0308]: mismatched types
   --> /home/amartin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rdkafka-0.25.0/src/config.rs:153:17
    |
150 |             rdsys::rd_kafka_conf_get(
    |             ------------------------ arguments to this function are incorrect
...
153 |                 buf.as_mut_ptr() as *mut i8,
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`
note: function defined here
   --> /home/amartin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rdkafka-sys-3.0.0+1.6.0/src/bindings.rs:922:12
    |
922 |     pub fn rd_kafka_conf_get(
    |            ^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `rdkafka` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...

### Steps to reproduce the behavior

Error occurs when building on my system.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,7 @@ rand_distr = "0.4.3"
 rand_xoshiro = "0.6.0"
 ratelimit = "0.7.1"
 redis = { version = "0.23.3", features = ["tokio-comp"] }
-rdkafka = { version = "0.25", features = ["cmake-build"] }
+rdkafka = { version = "0.34.0", features = ["cmake-build"] }
 ringlog = "0.3.1"
 rpcperf-dataspec = { path = "lib/dataspec" }
 serde = { workspace = true }

